spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaDiscoveryClientConfiguration.java

```diff
@@ -51,52 +52,100 @@
+        return new EurekaDiscoveryClient(client, clientConfig);
+    }
+
+    @Configuration(proxyBeanMethods = false)
+    @ConditionalOnProperty(value = "eureka.client.healthcheck.enabled", matchIfMissing = false)
+    protected static class EurekaHealthCheckHandlerConfiguration {
+
+        @Autowired(required = false)
+        private StatusAggregator statusAggregator = new SimpleStatusAggregator();
+
+        @Bean
+        @ConditionalOnMissingBean(HealthCheckHandler.class)
+        public EurekaHealthCheckHandler eurekaHealthCheckHandler() {
+            return new EurekaHealthCheckHandler(this.statusAggregator);
+        }
+
+    }
+
+    @Configuration(proxyBeanMethods = false)
+    @ConditionalOnClass(RefreshScopeRefreshedEvent.class)
+    protected static class EurekaClientConfigurationRefresher
+            implements ApplicationListener<RefreshScopeRefreshedEvent> {
+
+        private final Semaphore semaphore = new Semaphore(1);
+
+        @Autowired(required = false)
+        private EurekaClient eurekaClient;
+
+        @Autowired(required = false)
+        private EurekaAutoServiceRegistration autoRegistration;
+
+        public void onApplicationEvent(RefreshScopeRefreshedEvent event) {
+            try {
+                semaphore.acquire();
+                if (eurekaClient != null) {
+                    eurekaClient.getApplications();
+                }
+                if (autoRegistration != null) {
+                    // deregister instance
+                    this.autoRegistration.stop();
+                    // register instance
+                    this.autoRegistration.start();
+                }
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            } finally {
+                semaphore.release();
+            }
+        }
+
+    }
+
+    @Configuration(proxyBeanMethods = false)
+    protected static class DiscoveryClientConfiguration {
+
+        private final Semaphore semaphore = new Semaphore(1);
+
+        @Autowired
+        private ApplicationInfoManager applicationInfoManager;
+
+        @Autowired
+        private HealthCheckHandler healthCheckHandler;
+
+        @Autowired
+        private InstanceInfo instanceInfo;
+
+        void refreshInstanceInfo() {
+            try {
+                semaphore.acquire();
+                applicationInfoManager.refreshDataCenterInfoIfRequired();
+                applicationInfoManager.refreshLeaseInfoIfRequired();
+
+                InstanceStatus status;
+                try {
+                    // get instance status
+                    status = healthCheckHandler.getStatus(instanceInfo.getStatus());
+                } catch (Exception e) {
+                    logger.warn("Exception from healthcheckHandler.getStatus, setting status to DOWN", e);
+                    status = InstanceStatus.DOWN;
+                }
+
+                if (null != status) {
```

Comment:
The code checks for null correctly, but it would be more readable to use status != null instead of null != status. This aligns with common Java conventions and enhances clarity for future readers of the code.